lib/ioutil/log_writer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ioutil
+
+import (
+	"bufio"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/mailru/easyjson/jwriter"
+)
+
+// LogWriter wraps entries in a LogEntry struct with the
+// timestamp of the entry.
+type LogWriter struct {
+	w    io.Writer
+	prev []byte
+}
+
+// NewLogWriter wraps an io.WriteCloser in a LogWriter
+func NewLogWriter(w io.WriteCloser) *LogWriter {
+	return &LogWriter{w: w}
+}
+
+// Write takes an incoming byte slice and wraps it in a LogEntry,
+// adding a timestamp. It then serializes the LogEntry and writes the
+// serialized bytes to the underlying stream, adding a newline at the end for
+// human readability
+func (lw *LogWriter) Write(p []byte) (n int, err error) {
+
+	w := 0
```

Comment:
Nitpick.  w is also the member variable name.  It would be easy to accidentally mix the two in future code changes.